model/internal/data/protogen/metrics/v1/metrics.pb.go

```diff
@@ -2484,6 +2492,20 @@ func (m *HistogramDataPoint) MarshalToSizedBuffer(dAtA []byte) (int, error) {
 	_ = i
 	var l int
 	_ = l
+	if !m.Max.IsEmpty() {
+		i -= m.Max.Size()
+		if _, err := m.Max.MarshalTo(dAtA[i:]); err != nil {
+			return 0, err
+		}
+		dAtA[i] = 0x61
+	}
+	if !m.Min.IsEmpty() {
+		i -= m.Min.Size()
+		if _, err := m.Min.MarshalTo(dAtA[i:]); err != nil {
+			return 0, err
+		}
+		dAtA[i] = 0x59
+	}
```

Comment:
Is this manually written? How can we maintain this?

Reply:
Correct, the process here would be that it would have to be added back to the proto after `make genproto` is run. Any thoughts on how to best address this? I suppose we could add this marshal/unmarshal methods outside of the generated code, and remove the function as part of `genproto`?

I'm open to alternatives here.

Reply:
Can we automate the patching as part of `make genproto` step? Also extract the manually written as much as possible as a func and move to a regular (non-generated) file and call it from here (minimize the line count of manual code present in this auto-generated file).

Reply:
Also add a comment clearly saying where does the manual code in this file start and where it ends. This should help to locate the right place to manually patch after re-generation and if auto-patching fails.